sql/parser/sql.y

```diff
@@ -24,6 +24,7 @@ import "github.com/cockroachdb/cockroach/sql/privilege"
 %}
 
 %union {
+  boolean        bool
```

Comment:
Using `ival` for `opt_unique` was not ugly in my opinion, but I'm fine if your conscience needs type safety. Let's rename this to `bVal` or `boolVal` for consistency and place it below `ival`. `id` and `pos` are the primary fields in this union and belong at a place of prominence at the top.